### PR TITLE
Support for @layer CSS-rule

### LINKED ISSUE #14085: "Could not parse CSS stylesheet" when running tests with jest
### Describe the bug

After upgrading primeng 16.3.1 to 16.7.2 our jest tests fail with the following error:

```
Error: Could not parse CSS stylesheet
    at exports.createStylesheet (...node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/helpers/stylesheets.js:34:21)
    at HTMLStyleElementImpl._updateAStyleBlock (...node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/nodes/HTMLStyleElement-impl.js:68:5)
    at HTMLStyleElementImpl._attach (...node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/nodes/HTMLStyleElement-impl.js:19:12)
[...]
detail: '@layer primeng{.p-datatable{position:relative}.p-datatable>.p-datatable-wrapper{overflow:auto}.p-datatable-table{border-spacing:0px;width:100%}.p-datatable .p-sortable-column{cursor:pointer;-web
[...]
```

These problems seem to start with version 16.4.0 and to be caused by the introduction of CSS layers.

### Environment

Angular app using nx workspace

### Reproducer

_No response_

### Angular version

16.2.9

### PrimeNG version

16.7.2

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.4.0

### Browser(s)

_No response_

### Steps to reproduce the behavior

1. Install primeng 16.7.2
2. npm run test

### Expected behavior

Jest tests should still pass.

### PR DESCRIPTION
**Pull request by simpleman383**
Added support for `@layer` CSS-rule (CSS Layer Block)

Gotta fix [this issue](https://github.com/primefaces/primeng/issues/14085) and similar ones since `jsdom` utilizes this package.

### CODE REVIEW

File: spec/parse.spec.js [hunk 1]
```diff
@@ -1328,7 +1328,35 @@ var TESTS = [
 
 			return result;
 		})()
-	}
+	},
+	{
+		input: "@layer custom-layer { div { display: block; } }",
```
Comment (hasezoey) [inline spec/parse.spec.js:1407]: consider also adding a test for anonymous layers (layers without a name)

`@layer { div { display: block; } }`

Comment (simpleman383) [reply spec/parse.spec.js:1407]: sure, done - in this case the model has an empty layerName

File: lib/parse.js [hunk 2]
```diff
@@ -165,7 +166,15 @@ CSSOM.parse = function parse(token) {
...
-			} else if (token.indexOf("@supports", i) === i) {
+			} else if (token.indexOf("@layer", i) === i) {
+				state = "layerBlock"
+				layerBlockRule = new CSSOM.CSSLayerBlockRule();
+				layerBlockRule.__starts = i;
+				i += "layer".length;
+				buffer = "";
+				break;
+			} 
+			else if (token.indexOf("@supports", i) === i) {
```
Comment (hasezoey) [inline lib/parse.js]:
```suggestion
			}  else if (token.indexOf("@supports", i) === i) {
```

other style surrounding it has this style

File: lib/parse.js [hunk 3]
```diff
@@ -254,7 +263,18 @@ CSSOM.parse = function parse(token) {
...
+
+				if (parentRule) {
+					ancestorRules.push(parentRule);
+				}
+
+				currentScope = parentRule = layerBlockRule;
+				layerBlockRule.parentStyleSheet = styleSheet;
+				buffer = "";
+				state = "before-selector";
+			}	else if (state === "hostRule-begin") {
```
Comment (hasezoey) [inline lib/parse.js]:
```suggestion
			} else if (state === "hostRule-begin") {
```

**Review by hasezoey — APPROVED** [review]
Looks good to me, some minor style suggestions (according to surrounding code).
Please note that i am not involved with this project and dont know the other code other than this PR, i also am not involved with css parsing in general.

**Comment by simpleman383** [comment]
Actually, seems like there was no code style policy at all even before I touched the code for the first time 😁

**Comment by simpleman383** [comment]
Fixing code styling problem would bring A LOT of changes among different files (even the files irrelevant to the issue). Maybe the repo needs some refactoring in general...

**Comment by smitev** [comment]
What is the status of this PR?
@YunFeng0817 can we merge it?

**Comment by tijmenvangurp** [comment]
Any plans to get @layer support in? I can see this pr is  open for a while, anything blocking to get it merged?

**Comment by smitev** [comment]
@YunFeng0817 it's been some time without attention to this PR.
can we get your feedback on it?

**Review by YunFeng0817 — APPROVED** [review]

**Pull request merged**